ui/app/mainui/AppMain.qml

```diff
@@ -158,8 +160,21 @@ Item {
         onOpenEditDisplayNamePopup: Global.openPopup(displayNamePopupComponent)
 
         onOpenPopupRequested: {
+            if (activePopupComponents.includes(popupComponent)) {
+                return;
+            }
+
             const popup = popupComponent.createObject(appMain, params);
             popup.open();
+
+            activePopupComponents.push(popupComponent);
+
+            popup.closed.connect(() => {
```

Comment:
Do we need to disconnect that at some point or the popup will get garbage collected?

Reply:
Yes, connection gets destroyed after emitter object is dead